jdk398/Shadow_of_Knight_Solution.cpp

```diff
@@ -0,0 +1,163 @@
+        string bomb_dir; 
+        cin >> bomb_dir; cin.ignore();
+
+        cur_batx = batx;
+        cur_baty = baty;
+        
+        // This "if block" should only run if the game is in the **first** round. This is for the second to last level because without it the program will fail, however even with this the program will pass on all other levels
+        //
+        // Unfortunately I couldn't think of another way to solve this problem without this quick fix. 
+        if (count == 0){
+            batx = 0;
+            baty = h/5;
+            count++;
+        }
+        // For each of these if, else if statements the direction is checked and the limit for the opposite direction is set. This is so that Batman doesn't accidentally go backwards in future rounds.
+        if (bomb_dir.find("U") != string::npos){
+            // here I assign my lower limit to be one space below where batman currently is. This helps me in later steps so when I'm calculating moves I don't accidentally move into an area I should have already ruled out.
+            lwr_lim = cur_baty+1;
+
+            // This assignment is necessary because the integer division below it will not give a position of 0, so if you're at y position 1 and the bomb is up the bomb is at position 0;
+            if(baty == 1){
+                baty = 0;
+            }
+            else{
+                // Here I use my upper limit to help find how far up I need to go. I don't want to move past where I've already been, so I subract my current position by the farthest up I've been and then divide by two to get the point in the middle.
+                baty -= (baty - upr_lim)/2;
+                // It's also important to note that I use a divide by 2 because I feel on average picking the point in the middle will cut out the most possible positions.
+            }
+        }
+        // The other three definitions are pretty similar to the first one, first we set the limit for the adjacent line of coordinates opposite to the direction the bomb is in, then we set the next coordinate as halfway between where we currently are and the limit in the direction of the bomb.
+        if (bomb_dir.find("R") != string::npos){
+            lft_lim = cur_batx-1;
+
+            batx += (rgt_lim - batx)/2;
+        }
+        if (bomb_dir.find("D") != string::npos){
+            upr_lim = cur_baty;
+
+            baty += (lwr_lim - baty)/2;
+        }
+        if (bomb_dir.find("L") != string::npos){
+            rgt_lim = cur_batx+1;
+
+            if(batx == 1){
+                batx = 0;
+            }
+            else{
+                batx -= (batx - lft_lim)/2;
+            }
+        }
+        // Below is an example of how I did my boolean operations. I looked for the exact string for each possible output. This was not the best way to do this because it took up way too much space.
```

Comment:
I'm confused -- this is old code you decided not to use?

Reply:
Yes, I updated the comment above it to try and make it clearer. Do you think it would be better to insert this above with the bulk of my explanation?

Reply:
I changed how I represented the code, putting it inside of the ` .. code::`  syntax. Hopefully this made it clear that this is an example of how I solved the problem before using the `.find()` functionality of `string`